src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -233,6 +234,16 @@ private static ScheduledThreadPoolExecutor initSyncExecutor()
 
     private volatile double crcCheckChance;
 
+    public static Iterable<SSTableReader> selectOnlyBigTableReaders(Iterable<? extends SSTableReader> readers)
+    {
+        return Iterables.transform(Iterables.filter(readers, tr -> tr.descriptor.formatType == SSTableFormat.Type.BIG), SSTableReader.class::cast);
```

Comment:
This `transform` is one of the reasons the `Ref` hierarchy should not be using wildcards in the returned values.